tree/ntuple/v7/inc/ROOT/RPageStorageDaos.hxx

```diff
@@ -79,6 +81,66 @@ struct RDaosNTupleAnchor {
    static std::uint32_t GetSize();
 };
 
+// clang-format off
+/**
+\class ROOT::Experimental::Detail::RDaosContainerNTupleLocator
+\ingroup NTuple
+\brief Helper structure concentrating the functionality required to locate an RNTuple within a DAOS container.
+It includes a hashing function that converts the RNTuple's name into a 32-bit identifier; this value is used to specify
+the subspace for the RNTuple among all objects in the container. A zero-value hash is reserved and disallowed.
+The identifier is used to read the metadata (anchor / header / footer, pagelists excluded) from the store in the
+`GetNTupleDescriptorBuilder()` method. A copy of the anchor is stored in `fAnchor`.
+*/
+// clang-format on
+struct RDaosContainerNTupleLocator {
+   std::string fName{};
+   std::uint32_t fHashedName{};
+   ntuple_index_t fIndex{};
+   std::optional<ROOT::Experimental::Detail::RDaosNTupleAnchor> fAnchor;
+   static const ntuple_index_t kReservedHash = 0;
+
+   RDaosContainerNTupleLocator() = default;
+   explicit RDaosContainerNTupleLocator(const std::string &ntupleName)
+      : fName(ntupleName), fHashedName(Hash(ntupleName))
+   {
+      fIndex = fHashedName;
+   };
+
+   ntuple_index_t GetIndex() const { return fIndex; };
+   static ntuple_index_t Hash(const std::string &ntupleName)
+   {
+      // Convert string to numeric representation via `std::hash`.
+      std::size_t h = std::hash<std::string>{}(ntupleName);
+      // Fold `std::size_t` bits into 32-bit using `boost::hash_combine()` algorithm and magic number.
+      auto seed = static_cast<uint32_t>(h >> 32);
+      seed ^= static_cast<uint32_t>(h & 0xffffffff) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+      ntuple_index_t hash = static_cast<ntuple_index_t>(seed);
+
+      return (hash == kReservedHash) ? kReservedHash + 1 : hash;
+   }
+
+   int GetNTupleDescriptorBuilder(RDaosContainer &cont, RNTupleDecompressor &decompressor,
+                                  RNTupleDescriptorBuilder &builder);
```

Comment:
Given that it takes a reference to a `RNTupleDescriptorBuilder`, maybe rename as follows:
```suggestion
   int InitRNTupleDescriptorBuilder(RDaosContainer &cont, RNTupleDecompressor &decompressor,
                                  RNTupleDescriptorBuilder &builder);
```